src/System.Memory/src/System/Buffers/SequenceReader.cs

```diff
@@ -75,24 +77,24 @@ public ReadOnlySpan<T> UnreadSpan
         /// <summary>
         /// The total number of <typeparamref name="T"/>'s processed by the reader.
         /// </summary>
-        public long Consumed { get; private set; }
+        public long Consumed { readonly get; private set; }
 
         /// <summary>
         /// Remaining <typeparamref name="T"/>'s in the reader's <see cref="Sequence"/>.
         /// </summary>
-        public long Remaining => Length - Consumed;
+        public readonly long Remaining => Length - Consumed;
 
         /// <summary>
         /// Count of <typeparamref name="T"/> in the reader's <see cref="Sequence"/>.
         /// </summary>
-        public long Length
+        public readonly long Length
         {
             get
             {
-                if (_length < 0)
+                if (_length == -1)
```

Comment:
Nit: Why change this to `== -1`? Comparing with 0 tends to be tiny bit faster.

Reply:
Accident :) Fixing